### ISSUE TITLE
Error When compiling Ruby Application

### ISSUE THREAD

**Issue by enyachoke** [flynn-archive/slugbuilder#2]
When I run `git archive master | docker run -i -a stdin -a stdout flynn/slugbuilder`
I get this error. What could be the problem

```
-----> Ruby app detected
-----> Compiling Ruby/Rails
       !
       !     Command: 'set -o pipefail; curl --fail --retry 3 --retry-delay 1 --connect-timeout 3 --max-time 20 https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/ruby-2.0.0.tgz -s -o - | tar zxf -' failed unexpectedly:
       !
       !     gzip: stdin: invalid compressed data--format violated
       !     tar: Unexpected EOF in archive
       !     tar: Unexpected EOF in archive
       !     tar: Error is not recoverable: exiting now
```

**Comment by progrium** [comment]
This is a problem happening in the Ruby buildpack as it downloads Ruby from
S3. It could be temporary, the link works for me now.

On Fri, Nov 1, 2013 at 1:29 PM, Emmanuel Nyachoke
notifications@github.comwrote:

> When I run git archive master | docker run -i -a stdin -a stdout
> flynn/slugbuilder
> I get this error. What could be the problem
> 
> -----> Ruby app detected
> -----> Compiling Ruby/Rails
>        !
>        !     Command: 'set -o pipefail; curl --fail --retry 3 --retry-delay 1 --connect-timeout 3 --max-time 20 https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/ruby-2.0.0.tgz -s -o - | tar zxf -' failed unexpectedly:
>        !
>        !     gzip: stdin: invalid compressed data--format violated
>        !     tar: Unexpected EOF in archive
>        !     tar: Unexpected EOF in archive
>        !     tar: Error is not recoverable: exiting now
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flynn/slugbuilder/issues/2
> .

## 

Jeff Lindsay
http://progrium.com

**Comment by enyachoke** [comment]
Ot maybe it is my connection let me try again. Thanks

**Comment by enyachoke** [comment]
Closing,

**Comment by codedependant** [comment]
I'm experiencing the same issue, but it doesn't seem to be temporary for me.

Any idea how I can resolve this?

**Comment by codedependant** [comment]
just out of interest, I can download ruby-2.0.0.tgz and run `tar zxf` on it successfully. so it seems to be something about the command `set -o pipefail; curl --fail --retry 3 --retry-delay 1 --connect-timeout 3 --max-time 20 https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/ruby-2.0.0.tgz -s -o - | tar zxf -` that is the issue.

**Comment by progrium** [comment]
I'd take it up with the ruby buildpack maintainer.

On Wed, Nov 20, 2013 at 11:27 AM, Che notifications@github.com wrote:

> just out of interest, I can download ruby-2.0.0.tgz and run tar zxf on it
> successfully. so it seems to be something about the command set -o
> pipefail; curl --fail --retry 3 --retry-delay 1 --connect-timeout 3
> --max-time 20
> https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/ruby-2.0.0.tgz-s -o - | tar zxf -that is the issue.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/flynn/slugbuilder/issues/2#issuecomment-28909761
> .

## 

Jeff Lindsay
http://progrium.com

**Comment by codedependant** [comment]
yeah, just did that.

if anyone's interested:
https://github.com/heroku/heroku-buildpack-ruby/pull/175
